File: workspaces/001.fundamentals/code/01.order_service.rb
class OrderService

  # BAD: The method signature is huge and hard to read.
  # The procedure is "obsessed" with Strings and ints.
  def place_order(customer_name, customer_email,
                  street, city, zip_code,
                  pizza_size, toppings_count)

    # Step 1: Validate Data (Clutters the procedure)
    unless customer_email.include?("@") && customer_email.include?(".")
      raise ArgumentError, "Invalid email format"
    end
    unless zip_code.length == 5
      raise ArgumentError, "Invalid zip code"
    end

    # Step 2: Calculate Price (Business logic mixed with control flow)
    price = case pizza_size
            when "Small" then 10.00
            when "Medium" then 12.00
            when "Large" then 14.00
            else 0.0
            end

    price += toppings_count * 1.50

    # Step 3: Print Receipt
    puts "Order for: #{customer_name} (#{customer_email})"
    puts "Ship to: #{street}, #{city} #{zip_code}"
    puts "Total: $#{price}"
  end
end

File: workspaces/001.fundamentals/code/01.order_service.java
public class OrderService {

    // BAD: The method signature is huge and hard to read.
    // The procedure is "obsessed" with Strings and ints.
    public void placeOrder(String customerName, String customerEmail, 
                           String street, String city, String zipCode, 
                           String pizzaSize, int toppingsCount) {

        // Step 1: Validate Data (Clutters the procedure)
        if (!customerEmail.contains("@") || !customerEmail.contains(".")) {
            throw new IllegalArgumentException("Invalid email format");
        }
        if (zipCode.length() != 5) {
            throw new IllegalArgumentException("Invalid zip code");
        }

        // Step 2: Calculate Price (Business logic mixed with control flow)
        double price = 0.0;
        if (pizzaSize.equals("Small")) price = 10.00;
        else if (pizzaSize.equals("Medium")) price = 12.00;
        else if (pizzaSize.equals("Large")) price = 14.00;
        
        price += (toppingsCount * 1.50);

        // Step 3: Print Receipt
        System.out.println("Order for: " + customerName + " (" + customerEmail + ")");
        System.out.println("Ship to: " + street + ", " + city + " " + zipCode);
        System.out.println("Total: $" + price);
    }
}